src/js/gamestate.js

```diff
@@ -124,6 +146,10 @@ var gameState = function($q, notificationService, $filter, UPGRADES, GainCalcula
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/seiyria/c/file/626050834/issues/source?bid=2277746&fileBranchId=2301504#l146)